### DISCUSSION TITLE
Renovate 36.0.0 "Invalid configuration option: docker"

### DISCUSSION THREAD

**Discussion by aarongoldenthal** [renovatebot/renovate#23146] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Local npm package, renovate/renovate:36.0.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Running `renovate-config-validator` locally with the v36.0.0 npm package, or the renovate/renovate:36.0.0 container image, identifies a config migration is necessary, showing the old and new configs. When trying the new config, it fails with:

```sh
       "file": ".gitlab/renovate.json",
       "errors": [
         {
           "topic": "Configuration Error",
           "message": "Invalid configuration option: docker"
         }
       ]
```

Trying this with a minimal config, it occurs with the following as long as the `docker` property is present (and passes when it is not).

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:recommended", "docker:enableMajor"],
  "docker": { "pinDigests": true }
}
```

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by secustor** [comment]
Languages have been removed in v36.0.0, rather use `matchCategories` and `packageRules`

```json
{
 "packageRules": [
    {
       "matchCategories": ["docker"],
       "pinDigests": true
    }
  ]
}
```

**Reply by aarongoldenthal** [reply]
Then the issue seems to be the recommended new config that Renovate provides, since it still includes `docker` and `npm` (at least in my configs).

**Reply by aarongoldenthal** [reply]
As an example: https://gitlab.com/gitlab-ci-utils/pa11y-ci-reporter-runner/-/jobs/4593029892

**Reply by secustor** [reply]
The page updated 2 minutes ago automatically https://github.com/renovatebot/renovatebot.github.io/actions/runs/5458225239/jobs/9933088426
and with it the schema.

**Reply by viceice** [reply]
@rarkins should we add some compabillity migration? 🤔

**Reply by rarkins** [reply]
Yes, but I thought it already had?

**Reply by secustor** [reply]
Not top level configurations https://github.com/renovatebot/renovate/pull/16534#discussion_r1092796768

**Reply by rarkins** [reply]
Oh no.. I misunderstood that. We _must_ migrate those top level config options or it will break thousands in the hosted app alone

**Reply by rarkins** [reply, marked as answer]
- https://github.com/renovatebot/renovate/issues/23155

**Comment by rarkins** [comment]
@aarongoldenthal can you retry v36.0.3?

**Reply by aarongoldenthal** [reply]
@rarkins I tried 36.0.3 (and the latest 36.1.1) and they successfully migrated `docker` and `python` to a new top-level `packageRules`. Is the `npm` manger still valid at the top level (or any of the managers)? I thought it used to be in the config docs (at least that's how I built the config), but don't see it there now, and it was not migrated. It said `Config validated successfully`, so I assume that's expected.

**Reply by rarkins** [reply]
I think `npm` is not a language, it's a manager/datasource, so it's handled differently.

**Reply by secustor** [reply]
> I think npm is not a language, it's a manager/datasource, so it's handled differently.

☝️ 

Top level manager configurations are still valid in 36.x.x
